types/address.go

```diff
@@ -237,39 +249,30 @@ func (aa AccAddress) Bytes() []byte {
 	return aa
 }
 
-// AccAddress.String() is expensive and if unoptimized dominantly showed up in profiles,
-// yet has no mechanisms to trivially cache the result given that AccAddress is a []byte type.
-// With the string interning below, we are able to achieve zero cost allocations for string->[]byte
-// because the Go compiler recognizes the special case of map[string([]byte)] when used exactly
-// in that pattern. See https://github.com/cosmos/cosmos-sdk/issues/8693.
-var addMu sync.Mutex
-var addrStrMemoize = make(map[string]string)
-
 // String implements the Stringer interface.
-func (aa AccAddress) String() (str string) {
-	addMu.Lock()
-	defer addMu.Unlock()
-
-	if str, ok := addrStrMemoize[string(aa)]; ok {
-		return str
+func (aa AccAddress) String() string {
+	var key = string(aa)
```

Comment:
This burns an allocation from []byte->str, we need to fix this and we have code for it :-)

Reply:
I was thinking about it, but I didn't want to introduce unsafe code here.
Nevertheless, as you see in the benchmark, the new code is doing slightly less allocations and has same speed.

Reply:
I created a PR for moving the unsafe convertion code to an internal `conv` package: https://github.com/cosmos/cosmos-sdk/pull/8733